### ISSUE TITLE
MD5 on reference sequences that are too short

### ISSUE THREAD

**Issue by jkbonfield** [enasequence/cramtools#30]
A single read file consisting of an entry 9 bases long causes an error.

```
 jkb@seq3a[htslib/test] cat a.sam
@SQ     SN:xx   LN:20
a1      0       xx      1       50      9M      *       0       0       AAAAAAAAA       *
jkb@seq3a[htslib/test] java -Xmx4000m -jar ~/work/cram/cramtools/cramtools-3.0.jar cram -R xx.fa -I a.sam -O _.cram
Exception in thread "main" java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:606)
        at net.sf.cram.CramTools.invoke(CramTools.java:93)
        at net.sf.cram.CramTools.main(CramTools.java:123)
Caused by: java.lang.ArrayIndexOutOfBoundsException
        at java.lang.System.arraycopy(Native Method)
        at java.util.Arrays.copyOfRange(Arrays.java:2551)
        at net.sf.cram.structure.Slice.setRefMD5(Slice.java:272)
        at net.sf.cram.Bam2Cram.main(Bam2Cram.java:372)
        ... 6 more
```

The cause appears to be the use of "int shoulder = 10" on line 268 of src/main/java/net/sf/cram/structure/Slice.java.

What is the purpose of this shoulder? Is it to create a slightly larger pos/span range than is used by the reads within the slice?

**Comment by vadimzalunin** [comment]
clumsy debugging technique: diagnostics that introduces new bugs

fix: 05c5fd5f02802ff2e09f1c6d489fc8dceb292b34

**Comment by jkbonfield** [comment]
Checked and it works for me now. Thanks.
